### ISSUE TITLE
New Rule: no-trailing-spaces

### ISSUE THREAD

**Issue by juhwano** [eslint/markdown#612]
### What rule do you want?

`no-trailing-spaces` - Disallow trailing whitespace at the end of lines

### What category of rule is this?

New Rule

### Example code

```md
This line has trailing spaces   
Another line with tabs	
Clean line without trailing whitespace
```

### What should the rule do?

Report an error when a line contains trailing whitespace (spaces or tabs) at the end.

**Options:**
- `skipBlankLines` (boolean, default: `false`): Allow trailing spaces on blank lines
- `ignoreCodeBlocks` (boolean, default: `true`): Don't check inside fenced code blocks

### Prior Art

- [markdownlint MD009](https://github.com/DavidAnson/markdownlint/blob/main/doc/md009.md) - Trailing spaces
- [remark-lint no-final-newline](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-final-newline)

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

Trailing whitespace can cause unintended line breaks in rendered markdown (two spaces = `<br>`) and is generally considered noise in version control diffs.
